README.md

```diff
@@ -61,71 +62,143 @@ ragel >= 6.10
 # Installation
 
 - Clone the repo
- ```
+
+```
 git clone --recurse-submodules git@github.com:NilFoundation/proof-market-toolchain.git
 cd proof-market-toolchain
 ```
+
+# Building
+
 You need to build the proof generator only if you wish to generate proofs.
 
-#### Building
+## Building on host
+
 ```
 mkdir build
 cd build
-cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/bin/clang-12 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-12 ..
-# Single-threaded version
+cmake -G "Unix Makefiles" \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_C_COMPILER=/usr/bin/clang-12 \
+  -DCMAKE_CXX_COMPILER=/usr/bin/clang++-12 \
+  ..
+# Single-threaded version (recommended)
 cmake --build . -t proof-generator
 # Multi-threaded version
 cmake --build . -t proof-generator-mt
 ```
 
-Building with Docker
+## Building with Docker
+
+First, build the toolchain image with all required dependencies.
+It can be reused multiple times for building and running proof market
+binaries.
+
 ```
-docker build -t proof-market-toolchain-docker .  
-docker run -it --mount type=bind,src="$(PWD)",target=/proof-market-toolchain proof-market-toolchain-docker  sh ./build.sh
+docker build -t proof-market-toolchain .
+```
+
+Next, run a container and build the proof market in it.
```

Comment:
It is not a Proof market. It is the proof market toolchain

Reply:
fixed